# Iris Flower Clasification

This project aim to classify the Iris flowers into their respective species using machine learning techniques.


In [2]:
from sklearn.datasets import load_iris

In [3]:
iris = load_iris()

In [4]:
import pandas as pd

In [5]:
df = pd.DataFrame(iris.data, columns=iris.feature_names)

In [6]:
df["species"] = iris.target

In [7]:
df.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   species  
0        0  
1        0  
2        0  
3        0  
4        0

In [8]:
df.shape

(150, 5)

In [9]:
df.columns

Index(['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)',
       'petal width (cm)', 'species'],
      dtype='object')

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   species            150 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 6.0 KB


In [11]:
df.describe

<bound method NDFrame.describe of      sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                  5.1               3.5                1.4               0.2   
1                  4.9               3.0                1.4               0.2   
2                  4.7               3.2                1.3               0.2   
3                  4.6               3.1                1.5               0.2   
4                  5.0               3.6                1.4               0.2   
..                 ...               ...                ...               ...   
145                6.7               3.0                5.2               2.3   
146                6.3               2.5                5.0               1.9   
147                6.5               3.0                5.2               2.0   
148                6.2               3.4                5.4               2.3   
149                5.9               3.0                5.1               1.8   

     species  
0          0  
1          0  
2          0  
3          0  
4          0  
..       ...  
145        2  
146        2  
147        2  
148        2  
149        2  

[150 rows x 5 columns]>

In [12]:
df["species"].value_counts()

species
0    50
1    50
2    50
Name: count, dtype: int64

## Exploratory Data Analysis

The dataset contains total 150 samples with 4 numerical features related to sepal and petal measurements. The target variable 'species' has 3 classes and is evenly distributed across the dataset. No missing values were observed, making the dataset clean and suitable for classification

In [13]:
x = df.drop("species", axis=1)
y = df["species"]

In [14]:
x.shape, y.shape

((150, 4), (150,))

In [15]:
from sklearn.model_selection import train_test_split

In [17]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42 )

## Data Preparation

The feature set and target variable were seperated to prepare the data modeling. The datastet was then split into training and testing sets, with 80% of the data used for training and 20% for testing. This helps evaluate the model unseen data.

In [18]:
from sklearn.linear_model import LogisticRegression

In [20]:
model = LogisticRegression(max_iter=200)

In [21]:
model.fit(x_train, y_train)

LogisticRegression(max_iter=200)

In [22]:
y_pred = model.predict(x_test)

In [23]:
from sklearn.metrics import accuracy_score

In [24]:
accuracy = accuracy_score(y_test, y_pred)
accuracy

1.0

In [25]:
from sklearn.metrics import classification_report

In [26]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        10
           1       1.00      1.00      1.00         9
           2       1.00      1.00      1.00        11

    accuracy                           1.00        30
   macro avg       1.00      1.00      1.00        30
weighted avg       1.00      1.00      1.00        30



## Model Training and Evaluation

A Logistic Regression model was trained using the prepared training dataset. The model was evaluated on the test set to assess its performnce on unseen data. The result show that model is able to classify iris species with complete accuracy, indicating separation between classes based on the given features.

## Conclusion

In this project, I worked on classifying iris flowers into different species using their sepal and petal measurements. I started by understanding and exploring the dataset, followed by preparing the data so it could be used effectively for model training.

A Logistic Regression model was then trained and tested on the data. The model performed well on the test set, achieving high accuracy, which shows that the given features are useful in distinguishing between the different iris species.

Overall, this project helped me strengthen my understanding of the complete machine learning workflow, including data exploration, preprocessing, model training, and evaluation for a classification problem.